solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2061,13 +2066,16 @@ public void reload(String name, UUID coreId) {
           RefCounted<IndexWriter> iwRef = core.getSolrCoreState().getIndexWriter(null);
           if (iwRef != null) {
             IndexWriter iw = iwRef.get();
-            // switch old core to readOnly
-            core.readOnly = true;
```

Comment:
I suppose doing this before was to protect changes happening after the commit sneaking it.  Why did you move it?

Reply:
Hum, good point. I moved it because I didn't realize your point and UpdateHanlder.commit() checks the code is not readonly, otherwise stops with an exception.
Actually, with your nice idea of an enum in CommitUpdateCommand, the intent would be clear and commit() would not reject if the core is closed after the commit. And we could restore the core.readonly = true at the right location here.